### PR TITLE
Kubelet: Remove runonce mode

### PR DESCRIPTION
**Pull request by HirazawaUi**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Runonce mode stopped working in https://github.com/kubernetes/kubernetes/pull/90377, because the `cpuManager` is not initialized in runonce, when calling [activePods()](https://github.com/kubernetes/kubernetes/blob/44b2c363f2ff05554d223ce7a883388ec32576c3/pkg/kubelet/cm/cpumanager/cpu_manager.go#L364) will cause nil pointer and kubelet panic. The milestone of https://github.com/kubernetes/kubernetes/pull/90377 is v1.19。

Therefore, we have chosen not to phase it out gradually, but to remove it directly.

ref: https://github.com/kubernetes/enhancements/pull/4736#discussion_r1688808894
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Removed support for the kubelet `--runonce` mode.
If you specify the kubelet command line flag `--runonce`, this is an error.
Setting `runOnce` in a kubelet configuration file is also an error, and specifying any
value for that configuration option is now deprecated.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by HirazawaUi** [comment]
Do we need to merge it into v1.31?

**Comment by kannon92** [comment]
We are past code freeze so we would need an exception for 1.31.

File: pkg/kubelet/apis/config/types.go [hunk 1]
```diff
@@ -281,6 +281,7 @@ type KubeletConfiguration struct {
 	ResolverConfig string
 	// RunOnce causes the Kubelet to check the API server once for pods,
 	// run those in addition to the pods specified by static pod files, and exit.
+	// Deprecated: no longer has any effect.
```
Comment (yujuhong) [inline pkg/kubelet/apis/config/types.go:284]: "Deprecated" seems to suggest that it still works. 
BTW, If it's still in the API, we can error out if it's set.

Comment (HirazawaUi) [reply pkg/kubelet/apis/config/types.go:284]: Added validation in `ValidateKubeletConfiguration` to check if the runonce mode is set. If it is set, the kubelet will stop running and output an error log.

**Comment by sftim** [comment]
Changelog suggestion:
```diff
-Remove support for the Runonce mode and mark the runonce option in `KubeletConfiguration` as deprecated.
+Removed support for the kubelet `--runonce` mode.
+The kubelet command line flag `--runonce` no longer has any effect, and using it is deprecated.
+Setting `runOnce` in a kubelet configuration file also has no effect, and the configuration option is also deprecated.
```

Is this right? Or does the `--runonce` option now trigger failure on run? If so, we should summarize that effect instead.

**Comment by bart0sh** [comment]
/lgtm

/assign @SergeyKanzhelev @mrunalp @derekwaynecarr 
for approval

**Comment by SergeyKanzhelev** [comment]
sorry I lost track of this one. If you can rebase fast, this may be OK to merge in 1.32

**Comment by HirazawaUi** [comment]
> sorry I lost track of this one. If you can rebase fast, this may be OK to merge in 1.32

rebased.

**Comment by mrunalp** [comment]
:+1:

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Comment by SergeyKanzhelev** [comment]
/assign @thockin 

@thockin, we discussed that removal of this do not require KEP as the feature is broken for a few releases now. @mrunalp thumbed up above.

**Pull request merged**